django/contrib/auth/views.py

```diff
@@ -442,11 +445,21 @@ class PasswordResetConfirmView(PasswordContextMixin, FormView):
     @method_decorator(never_cache)
     def dispatch(self, *args, **kwargs):
         assert 'uidb64' in kwargs and 'token' in kwargs
+        uidb64 = kwargs['uidb64']
+        token = kwargs['token']
 
         self.validlink = False
-        self.user = self.get_user(kwargs['uidb64'])
+        self.user = self.get_user(uidb64)
+
+        if token == INTERNAL_RESET_TOKEN:
+            token = self.request.session[INTERNAL_RESET_TOKEN]
+        else:
+            self.request.session[INTERNAL_RESET_TOKEN] = token
```

Comment:
yeah, `request.session.get` would return none for the token and this wouldn't pass the comparision (which would be perfectly fine)